### PR TITLE
Automate hook smoke test

### PR DESCRIPTION
**Pull request by Klovely**
#### Describe Bug or Feature
Automate Hook smoke Test for below scenario :
1.  Test for queuejob hook
2.  Test for modifyjob hook
3.  Test for resvsub hook
4.  Test for movejob hook
5.  Test for runjobhook

#### Describe Your Change
Added test script to cover above tests

#### Attach Test and Valgrind Logs/Output
[complete_TestHookSmokeTest_latest_execution_logs_on_craysim_x78.txt](https://github.com/PBSPro/pbspro/files/3528390/complete_TestHookSmokeTest_latest_execution_logs_on_craysim_x78.txt)
[complete_TestHookSmokeTest_latest_execution_logs_on_non_Cray.txt](https://github.com/PBSPro/pbspro/files/3528391/complete_TestHookSmokeTest_latest_execution_logs_on_non_Cray.txt)
[latest_logs_actual_cray_with_new_test.txt](https://github.com/PBSPro/pbspro/files/3528392/latest_logs_actual_cray_with_new_test.txt)

### CODE REVIEW

File: test/tests/functional/pbs_hooksmoketest.py [hunk 1]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+        sub_dir = self.du.create_temp_dir(asuser=TEST_USER)
+        a = {'Resource_List.mppwidth': 1, 'Resource_List.walltime': 30}
+        j1 = Job(TEST_USER, a)
+        j1.create_script(self.scr)
+        jid1 = self.server.submit(j1, submit_dir=sub_dir)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
+        # As a user submit a job without requesting walltime
+        # Job is denied with the message
+        _msg = "qsub: No walltime specified. Master does not approve! ;o)"
+        sub_dir = self.du.create_temp_dir(asuser=TEST_USER1)
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: let us say submit_dir

File: test/tests/functional/pbs_hooksmoketest.py [hunk 2]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+        a = {'Resource_List.mppwidth': 1}
+        j2 = Job(TEST_USER1, a)
+        j2.create_script(self.scr)
+        try:
+            jid2 = self.server.submit(j2, submit_dir=sub_dir)
+        except PbsSubmitError, e:
+            self.assertEqual(
+                e.msg[0], _msg, msg="Did not get expected qsub err message")
+            self.logger.info("Got expected qsub err message as %s", e.msg[0])
+            pass
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: can we do away with this "pass"

File: test/tests/functional/pbs_hooksmoketest.py [hunk 3]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+        # As a user submit aother job requesting walltime and -V
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: another?

File: test/tests/functional/pbs_hooksmoketest.py [hunk 4]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+        sub_dir = self.du.create_temp_dir(asuser=TEST_USER1)
+        self.ATTR_V = 'Full_Variable_List'
+        api_to_cli.setdefault(self.ATTR_V, 'V')
+        a = {self.ATTR_V: None, 'Resource_List.mppwidth': 1,
+             'Resource_List.walltime': 30}
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: could you please explain the purpose of ATTR_V here.

File: test/tests/functional/pbs_hooksmoketest.py [hunk 5]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+            " encountered an exception. Please inform Admin"
+        try:
+            rid = self.server.submit(r)
+        except PbsSubmitError, e:
+            self.assertEqual(
+                e.msg[0], _msg,
+                msg="Did not get expected pbs_rsub err message")
+            self.logger.info(
+                "Got expected pbs_rsub err message as %s", e.msg[0])
+            pass
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: can we do away with this "pass"

File: test/tests/functional/pbs_hooksmoketest.py [hunk 6]
```diff
@@ -24,12 +24,21 @@ class TestHookSmokeTest(TestFunctional):
     """
     Hooks Smoke Test
     """
+    hook_name = "test_hook"
+
```
Comment (dinesh1306) [inline test/tests/functional/pbs_hooksmoketest.py]: please remove the extra line.

File: test/tests/functional/pbs_hooksmoketest.py [hunk 7]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        # qalter the job, qalter will fail with error
+        _msg = "qalter: Event is incompatible " + jid1
+        try:
+            self.server.alterjob(jid1, {ATTR_p: '5'})
+        except PbsAlterError, e:
+            self.assertEqual(
+                e.msg[0], _msg, msg="Did not get expected qalter err message")
+            self.logger.info("Got expected qalter err message as %s", e.msg[0])
+            pass
```
Comment (dinesh1306) [inline test/tests/functional/pbs_hooksmoketest.py]: please show the error message handled in except block.

File: test/tests/functional/pbs_hooksmoketest.py [hunk 8]
```diff
@@ -206,6 +215,251 @@ def test_delete_hook(self):
...
+        s = repr(sel)
+else:
+        e.accept()"""
+
+        attrs = {'event': 'queuejob', 'enabled': 'True'}
+        rv = self.server.create_import_hook(self.hook_name, attrs, hook_body)
+        self.assertTrue(rv)
+        # As a user submit a job requesting walltime
+        sub_dir = self.du.create_temp_dir(asuser=TEST_USER)
+        a = {'Resource_List.mppwidth': 1, 'Resource_List.walltime': 30}
```
[the remark this answers is not in the document]
Comment (bhroam) [reply test/tests/functional/pbs_hooksmoketest.py]: I believe the mpp language is deprecated.  The translate hook is disabled by default.  Should we be testing the mpp language or should we be using select?

**Comment by Klovely** [comment]
Latest execution logs:-
[complete_TestHookSmokeTest_latest_execution_logs2_on_craysim.txt](https://github.com/PBSPro/pbspro/files/3533112/complete_TestHookSmokeTest_latest_execution_logs2_on_craysim.txt)
[complete_TestHookSmokeTest_latest_execution_logs2_on_non_Cray.txt](https://github.com/PBSPro/pbspro/files/3533113/complete_TestHookSmokeTest_latest_execution_logs2_on_non_Cray.txt)
[latest_logs2_cray_with_new_test.txt](https://github.com/PBSPro/pbspro/files/3533114/latest_logs2_cray_with_new_test.txt)

**Comment by Klovely** [comment]
Thanks @prakashcv13 and @dinesh1306 for comments.
Incorporated your comments in latest commits, please have a look.

**Comment by prakashcv13** [comment]
@bhroam - I do not have further comments, could you please take the PR forward.

**Comment by bhroam** [comment]
I didn't do a full review, but I don't think my comment was responded to.  Why are we testing the mpp language?  It's deprecated and we ship with the translate hook disabled.

**Comment by bhroam** [comment]
@prakashcv13 Lovely said she was going to remove the mpp resources.  Once that is done, please merge the PR.

**Comment by Klovely** [comment]
Latest execution logs : 
[complete_TestHookSmokeTest_latest_execution_logs3_on_craysim.txt](https://github.com/PBSPro/pbspro/files/3536790/complete_TestHookSmokeTest_latest_execution_logs3_on_craysim.txt)
[complete_TestHookSmokeTest_latest_execution_logs3_on_non_Cray.txt](https://github.com/PBSPro/pbspro/files/3536791/complete_TestHookSmokeTest_latest_execution_logs3_on_non_Cray.txt)

**Comment by prakashcv13** [comment]
merged #1264

**Pull request merged**